src/lv_core/lv_refr.c

```diff
@@ -317,39 +314,44 @@ void _lv_disp_refr_task(lv_task_t * task)
 /**
  * Join the areas which has got common parts
  */
-static void lv_refr_join_area(void)
-{
-    uint32_t join_from;
-    uint32_t join_in;
-    lv_area_t joined_area;
-    for(join_in = 0; join_in < disp_refr->inv_p; join_in++) {
-        if(disp_refr->inv_area_joined[join_in] != 0) continue;
-
-        /*Check all areas to join them in 'join_in'*/
-        for(join_from = 0; join_from < disp_refr->inv_p; join_from++) {
-            /*Handle only unjoined areas and ignore itself*/
-            if(disp_refr->inv_area_joined[join_from] != 0 || join_in == join_from) {
-                continue;
-            }
+static void lv_refr_join_area(void)  {
+      uint32_t join_from;
+      uint32_t join_in;
+      uint32_t diff; /* a badly named variable, counting the number of joined areas within an outer cycle */
+      lv_area_t joined_area;
 
-            /*Check if the areas are on each other*/
-            if(_lv_area_is_on(&disp_refr->inv_areas[join_in], &disp_refr->inv_areas[join_from]) == false) {
-                continue;
-            }
+      /* Sanity */
+      if (disp_refr->inv_p == 0) return;
 
-            _lv_area_join(&joined_area, &disp_refr->inv_areas[join_in], &disp_refr->inv_areas[join_from]);
+      for(join_in = 0; join_in < disp_refr->inv_p - 1; join_in++) {
 
-            /*Join two area only if the joined area size is smaller*/
-            if(lv_area_get_size(&joined_area) < (lv_area_get_size(&disp_refr->inv_areas[join_in]) +
-                                                 lv_area_get_size(&disp_refr->inv_areas[join_from]))) {
-                lv_area_copy(&disp_refr->inv_areas[join_in], &joined_area);
+          diff = 0;
+          /*Check all areas to join them in 'join_in'*/
+          for(join_from = join_in + 1; join_from < disp_refr->inv_p; join_from++) {
+
+              /* Move if there was some joining previously, to effectively remove the joined areas */
+              if (diff) {
+                lv_area_copy(&disp_refr->inv_areas[join_from - diff], &disp_refr->inv_areas[join_from]);
+              }
+
+              /*Check if the areas are on each other*/
+              if(lv_area_is_on(&disp_refr->inv_areas[join_in], &disp_refr->inv_areas[join_from]) == false) {
```

Comment:
The underscores are intended to be there.

Reply:
I saw your message on the forum. I'll fix the underscore issue - saves you the trouble of making another commit.

Reply:
Thanks.

Btw. what was the point of adding the underscores? 

JW

Reply:
The MicroPython binding, by default, converts every LVGL function into one accessible from MicroPython. There was a class of functions which no one was likely to make use of, so we added the `_` prefix as a way to tell MicroPython not to convert them. It also helps with autocompletion, since they won't be shown in the (already long) list of functions when one types `lv_`.

Reply:
I see, thanks.

JW